### ISSUE TITLE
Feature request: load music data from JSON / JS Objects

### ISSUE THREAD

**Issue by hunterloftis** [meenie/8bit.js#1]
For example:

``` js
var music = new EightBit();
var mario = {
  signature: [2, 2],
  tempo: 200,
  instruments: {
    left: 'square',
    right: 'square'
  },
  notes: [
    { instrument: 'right', type: 'note', time: 'quarter', data: 'E5, F#4' },
    { instrument: 'right', type: 'note', time: 'quarter', data: 'E5, F#4' },
    { instrument: 'right', type: 'rest', time: 'quarter' }
    // ... etc ...
  ]
};

music.load(mario);
music.play();
```

**Comment by meenie** [comment]
Yes, I like this a lot.  I should have done it this way in the first place :).  I'll put it together tonight.

Cheers,
cody

**Comment by meenie** [comment]
This is what I have so far.  The first version is the shorthand notation and second is more verbose with object properties.  Both are working.

``` javascript
gameMusic.load({
    timeSignature: [4, 4],
    tempo: 100,
    instruments: {
        rightHand: 'square',
        leftHand: 'sawtooth'
    },
    notes: {
        rightHand: [
            'E5, F#4|quarter|tie',
            'rest|quarter',
            'E5, F#4|quarter',
            'rest|quarter'
        ],
        leftHand: [
            'D3|quarter',
            'D3|quarter',
            'D3|quarter',
            'D3|quarter'
        ]
    }
});
```

More verbose version

``` javascript
gameMusic.load({
    timeSignature: [6, 8],
    tempo: 120,
    instruments: {
        rightHand: 'square',
        leftHand: 'sawtooth'
    },
    notes: {
        rightHand: [
            {
                type: 'note',
                pitch: 'E5, F#4',
                rhythm: 'quarter'
            }
        ],
        leftHand: [
            {
                type: 'note',
                pitch: 'D3',
                rhythm: 'quarter'
            }
        ]
    }
});
```

**Comment by hunterloftis** [comment]
Wow, nice - that was fast!

On Thursday, August 22, 2013, Cody Lundquist wrote:

> This is what I have so far. The first version is the shorthand notation
> and second is more verbose with object properties. Both are working.
> 
> gameMusic.load({
>     timeSignature: [4, 4],
>     tempo: 100,
>     instruments: {
>         rightHand: 'square',
>         leftHand: 'sawtooth'
>     },
>     notes: {
>         rightHand: [
>             'E5, F#4|quarter|tie',
>             'rest|quarter|',
>             'E5, F#4|quarter',
>             'rest|quarter'
>         ],
>         leftHand: [
>             'D3|quarter',
>             'D3|quarter',
>             'D3|quarter',
>             'D3|quarter'
>         ]
>     }});
> 
> More verbose version
> 
> gameMusic.load({
>     timeSignature: [4, 4],
>     tempo: 10,
>     instruments: {
>         rightHand: 'square',
>         leftHand: 'sawtooth'
>     },
>     notes: {
>         rightHand: [
>             {
>                 type: 'note',
>                 pitch: 'E5, F#4',
>                 rhythm: 'quarter'
>             }
>         ],
>         leftHand: [
>             {
>                 type: 'note',
>                 pitch: 'D3',
>                 rhythm: 'quarter'
>             }
>         ]
>     }});
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/meenie/8bit.js/issues/1#issuecomment-23086710
> .

## 
- from mobile

**Comment by meenie** [comment]
It was rather easy because of the way I put the library together in the first place.  I'll go ahead and commit this into the library and put in some documentation on how to use it.  Might have to be later today though.

**Comment by meenie** [comment]
I have released 0.1.0 with this new feature.  Closing off this issue.

Cheers,
cody
